llms/mlx_lm/utils.py

```diff
@@ -547,21 +544,25 @@ def load(
     """
     model_path = get_model_path(path_or_hf_repo)
 
-    model = load_model(model_path, lazy, model_config)
+    config = load_config(model_path)
+    config.update(model_config)
+
+    model = load_model(model_path, lazy, config)
     if adapter_path is not None:
         model = load_adapters(model, adapter_path)
         model.eval()
-    tokenizer = load_tokenizer(model_path, tokenizer_config)
+
+    tokenizer = load_tokenizer(model_path, tokenizer_config, model_config=config)
```

Comment:
Can you explain what the purpose of this change is? Does the `load_tokenizer` need the `config`?

Reply:
Currently, it appears that `load_tokenizer` indeed does not require the full `config` field information, only the `eos_token_id` field is needed.

At the time, the main consideration was that other parameters might be needed later, but it seems that the preliminary design for this part can be removed.